### PR TITLE
case 21932: the test-audio button causes a local audio-loopback rather than a server one

### PR DESCRIPTION
**Pull request by sethalves**
- the test-audio button causes a local audio-loopback rather than a server one

https://highfidelity.fogbugz.com/f/cases/21932/Test-your-voice-button-isn-t-working-correctly-when-Muted-or-in-Push-to-talk-mode

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18408/HighFidelity-Beta-PR15269-16f842b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18408/HighFidelity-Quest-Beta-PR15269-16f842b.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18408/HighFidelity-Beta-PR15269-16f842b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18408/HighFidelity-Beta-Interface-PR15269-16f842b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18408/HighFidelity-Beta-PR15269-16f842b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18408/HighFidelity-Beta-Interface-PR15269-16f842b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18410/HighFidelity-Beta-PR15269-5f8139a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18410/HighFidelity-Beta-Interface-PR15269-5f8139a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18410/HighFidelity-Beta-PR15269-5f8139a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18410/HighFidelity-Quest-Beta-PR15269-5f8139a.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18410/HighFidelity-Beta-PR15269-5f8139a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18410/HighFidelity-Beta-Interface-PR15269-5f8139a.exe

File: libraries/audio-client/src/AudioClient.cpp [hunk 1]
```diff
@@ -1052,7 +1052,7 @@ void AudioClient::setReverbOptions(const AudioEffectOptions* options) {
 void AudioClient::handleLocalEchoAndReverb(QByteArray& inputByteArray) {
     // If there is server echo, reverb will be applied to the recieved audio stream so no need to have it here.
     bool hasReverb = _reverb || _receivedAudioStream.hasReverb();
-    if (_muted || !_audioOutput || (!_shouldEchoLocally && !hasReverb)) {
+    if ((_muted && !_shouldEchoLocally) || !_audioOutput || (!_shouldEchoLocally && !hasReverb)) {
```
Comment (SamGondelman) [inline libraries/audio-client/src/AudioClient.cpp]: Isn't the `!_shouldEchoLocally` case already handled here?

Comment (sethalves) [reply libraries/audio-client/src/AudioClient.cpp]: The overall logic is changed, though you may need a Karnaugh map to really see it.  The point is to allow you to hear your own voice (for testing), even if you are muted.

Comment (SamGondelman) [reply libraries/audio-client/src/AudioClient.cpp]: wild!

**Pull request merged**